grammar/pat.lyg

```diff
@@ -1,18 +1,30 @@
 Pat =
   | Wild:"_"
+  | Rest:".."
   | Literal:{ minus:"-"? lit:LITERAL }
   // unstable(exclusive_range_pattern):
-  | Range:{ start:PatRangeValue ".." end:PatRangeValue }
-  | RangeInclusive:{ start:PatRangeValue { "..." | "..=" } end:PatRangeValue }
+  | Range:{
+    | start:PatRangeValue ".." end:PatRangeValue
+    // unstable(half_open_range_patterns):
+    | start:PatRangeValue ".." | ".." end:PatRangeValue
+  }
+  | RangeInclusive:{
+    | start:PatRangeValue { "..." | "..=" } end:PatRangeValue
+    // unstable(half_open_range_patterns):
+    | { "..." | "..=" } end:PatRangeValue
+  }
   | Binding:{ binding:Binding { "@" subpat:Pat }? }
   | Paren:{ "(" pat:Pat ")" }
   | Ref:{ "&" mutable:"mut"? pat:Pat }
   // unstable(box_patterns):
   | Box:{ "box" pat:Pat }
-  | Slice:{ "[" elems:SlicePatElem* %% "," "]" }
-  | Tuple:{ "(" fields:TuplePatField* %% "," ")" }
+  // unstable(or_patterns):
+  // FIXME(eddyb) find a way to express "2 or more" (like regex `{2,}`).
```

Comment:
Alternatively you can just explain this as a binary operator rather than a list.

Reply:
Fair, it's just weird to think of it as a binary operator because it's fully commutative and associative.